graphql.go

```diff
@@ -147,6 +147,9 @@ func (c *Client) runWithJSON(ctx context.Context, req *Request, resp interface{}
 		// return first error
 		return &gr.Errors[0]
 	}
+	if res.StatusCode != http.StatusOK {
```

Comment:
I think we should do this before even attempting to read the response body.

Reply:
Do graphql error responses not set an HTTP status code?

Reply:
yea, it's one thing that drives me a bit crazy about graphql... so maybe we do a status code check on a 5xx code and don't process the body?

Reply:
Yep. I moved the check to before we look at the body